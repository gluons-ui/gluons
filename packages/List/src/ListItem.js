import React from 'react';
import PropTypes from 'prop-types';

const ListItem = ({
  children,
  hasBorder,
  indented,
  lineColor,
  lineWidth,
  narrow,
  wide,
}) => {
  const indents = `${indented ? 'ph3' : ''}`;
  const borders = `${lineWidth === 0 ? '' : `bw${lineWidth}`}
                   ${hasBorder ? `bb b--${lineColor}` : 'bb b--transparent'}`;
  const spacing = `${narrow ? 'pv1' : `${wide ? 'pv3' : 'pv2'}`}`;
  return (
    <li className={`list-item ${indents} ${borders} ${spacing}`}>
      {children || null}
    </li>
  );
};

ListItem.propTypes = {
  lineColor: PropTypes.string,
  children: PropTypes.node,
  hasBorder: PropTypes.bool,
  indented: PropTypes.bool,
  lineWidth: PropTypes.number,
  narrow: PropTypes.bool,
  wide: PropTypes.bool,
};

ListItem.defaultProps = {
  lineColor: 'black',
  children: null,
  hasBorder: false,
  indented: false,
  lineWidth: 0,
  narrow: false,
  wide: false,
};

export default ListItem;
