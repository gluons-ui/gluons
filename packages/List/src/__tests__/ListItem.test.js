import React from 'react';
import { shallow } from 'enzyme';
import ListItem from '../ListItem';

describe('<ListItem />', () => {
  it('should exist', () => {
    expect(<ListItem />).toBeDefined();
  });

  it('should exist when called', () => {
    expect(shallow(<ListItem />).exists()).toEqual(true);
  });

  it('should render default with no props', () => {
    expect(shallow(<ListItem />)).toMatchSnapshot();
  });

  it('should render a gray border if the prop is called', () => {
    expect(shallow(
      <ListItem
        hasBorder
      >
        <span>I am a child</span>
      </ListItem>,
    )).toMatchSnapshot();
  });

  it('should render padding if indented', () => {
    expect(shallow(
      <ListItem
        indented
      >
        <span>I am a child</span>
      </ListItem>,
    )).toMatchSnapshot();
  });

  it('should render a custom border width', () => {
    expect(shallow(
      <ListItem
        hasBorder
        borderWidth={2}
      >
        <span>I am a child</span>
      </ListItem>,
    )).toMatchSnapshot();
  });
});
