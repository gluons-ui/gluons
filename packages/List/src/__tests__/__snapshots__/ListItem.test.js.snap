// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> should render a custom border width 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItem
    borderColor="black"
    borderWidth={2}
    hasBorder={true}
    indented={false}
  >
    <span>
      I am a child
    </span>
  </ListItem>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span>
        I am a child
      </span>,
      "className": "list-item pv3  bw2
                   bb b--black",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "I am a child",
      },
      "ref": null,
      "rendered": "I am a child",
      "type": "span",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span>
          I am a child
        </span>,
        "className": "list-item pv3  bw2
                   bb b--black",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "I am a child",
        },
        "ref": null,
        "rendered": "I am a child",
        "type": "span",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<ListItem /> should render a gray border if the prop is called 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItem
    borderColor="black"
    borderWidth={0}
    hasBorder={true}
    indented={false}
  >
    <span>
      I am a child
    </span>
  </ListItem>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span>
        I am a child
      </span>,
      "className": "list-item pv3  
                   bb b--black",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "I am a child",
      },
      "ref": null,
      "rendered": "I am a child",
      "type": "span",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span>
          I am a child
        </span>,
        "className": "list-item pv3  
                   bb b--black",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "I am a child",
        },
        "ref": null,
        "rendered": "I am a child",
        "type": "span",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<ListItem /> should render default with no props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItem
    borderColor="black"
    borderWidth={0}
    hasBorder={false}
    indented={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": null,
      "className": "list-item pv3  
                   bb b--transparent",
    },
    "ref": null,
    "rendered": null,
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": null,
        "className": "list-item pv3  
                   bb b--transparent",
      },
      "ref": null,
      "rendered": null,
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<ListItem /> should render padding if indented 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItem
    borderColor="black"
    borderWidth={0}
    hasBorder={false}
    indented={true}
  >
    <span>
      I am a child
    </span>
  </ListItem>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span>
        I am a child
      </span>,
      "className": "list-item pv3 ph3 
                   bb b--transparent",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "I am a child",
      },
      "ref": null,
      "rendered": "I am a child",
      "type": "span",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span>
          I am a child
        </span>,
        "className": "list-item pv3 ph3 
                   bb b--transparent",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "I am a child",
        },
        "ref": null,
        "rendered": "I am a child",
        "type": "span",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
